ccl/crosscluster/logical: TestLogicalJobResiliency failed

ccl/crosscluster/logical.TestLogicalJobResiliency [failed](https://mesolite.cluster.engflow.com/invocations/default/7bc7fba3-933e-47f7-b47b-8f7e7926f932?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [aa21162ed0577f1f22ca8938a3095bcecdaa09ca](https://github.com/cockroachdb/cockroach/commits/aa21162ed0577f1f22ca8938a3095bcecdaa09ca):


```
=== RUN   TestLogicalJobResiliency
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogicalJobResiliency1005768676
    test_log_scope.go:81: use -show-logs to present logs inline
    logical_replication_job_test.go:843: waiting for replication job 994765601824407553
    logical_replication_job_test.go:1016: condition failed to evaluate within 45s: from logical_replication_job_test.go:1024: waiting for stream ingestion job progress 1723648769.681518665,0 to advance beyond 1723648769.705997689,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogicalJobResiliency1005768676
--- FAIL: TestLogicalJobResiliency (66.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogicalJobResiliency.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41313

ccl/crosscluster/logical.TestLogicalJobResiliency [failed](https://tanzanite.cluster.engflow.com/invocations/default/7a7a99fd-85af-43d3-a1c6-bcd512c66564?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [575cdd4696dfcac8f311d1ea546683271102f73e](https://github.com/cockroachdb/cockroach/commits/575cdd4696dfcac8f311d1ea546683271102f73e):


```
=== RUN   TestLogicalJobResiliency
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogicalJobResiliency602760981
    test_log_scope.go:81: use -show-logs to present logs inline
    logical_replication_job_test.go:843: waiting for replication job 994979652668227585
    logical_replication_job_test.go:1016: condition failed to evaluate within 3m45s: from logical_replication_job_test.go:1024: waiting for stream ingestion job progress 1723714092.147421385,0 to advance beyond 1723714094.456007256,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogicalJobResiliency602760981
--- FAIL: TestLogicalJobResiliency (320.87s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogicalJobResiliency.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/crosscluster/logical.TestLogicalJobResiliency [failed](https://tanzanite.cluster.engflow.com/invocations/default/ae7ff879-d11b-4b83-8677-b447a098d77b?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [1993fc04b5116f20f4814d637c7ce87b003687e4](https://github.com/cockroachdb/cockroach/commits/1993fc04b5116f20f4814d637c7ce87b003687e4):


```
=== RUN   TestLogicalJobResiliency
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogicalJobResiliency557808735
    test_log_scope.go:81: use -show-logs to present logs inline
    logical_replication_job_test.go:858: waiting for replication job 995839861239513089
    logical_replication_job_test.go:1031: condition failed to evaluate within 45s: from logical_replication_job_test.go:1035: stream ingestion has not recorded any progress yet, waiting to advance pos 1723976607.707218020,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogicalJobResiliency557808735
--- FAIL: TestLogicalJobResiliency (70.71s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogicalJobResiliency.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/crosscluster/logical.TestLogicalJobResiliency [failed](https://tanzanite.cluster.engflow.com/invocations/default/329152ba-8c25-45ce-bc6b-1b0fe297b477?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [1993fc04b5116f20f4814d637c7ce87b003687e4](https://github.com/cockroachdb/cockroach/commits/1993fc04b5116f20f4814d637c7ce87b003687e4):


```
=== RUN   TestLogicalJobResiliency
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogicalJobResiliency2901528894
    test_log_scope.go:81: use -show-logs to present logs inline
    logical_replication_job_test.go:858: waiting for replication job 996112862871781377
    logical_replication_job_test.go:1031: condition failed to evaluate within 3m45s: from logical_replication_job_test.go:1035: stream ingestion has not recorded any progress yet, waiting to advance pos 1724059923.331396639,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogicalJobResiliency2901528894
--- FAIL: TestLogicalJobResiliency (311.07s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogicalJobResiliency.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/crosscluster/logical.TestLogicalJobResiliency [failed](https://tanzanite.cluster.engflow.com/invocations/default/f8114209-47be-42b1-8b8f-14731017cbaf?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [ba7c94a6ffe5f0a990000188ecfd3a23f7c8b4a6](https://github.com/cockroachdb/cockroach/commits/ba7c94a6ffe5f0a990000188ecfd3a23f7c8b4a6):


```
=== RUN   TestLogicalJobResiliency
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogicalJobResiliency3629262944
    test_log_scope.go:81: use -show-logs to present logs inline
    logical_replication_job_test.go:858: waiting for replication job 996383189412413441
    logical_replication_job_test.go:1031: condition failed to evaluate within 3m45s: from logical_replication_job_test.go:1035: stream ingestion has not recorded any progress yet, waiting to advance pos 1724142418.569472244,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogicalJobResiliency3629262944
--- FAIL: TestLogicalJobResiliency (309.77s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogicalJobResiliency.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/crosscluster/logical.TestLogicalJobResiliency [failed](https://tanzanite.cluster.engflow.com/invocations/default/50c3d031-4bf1-460e-ab47-37c71b820ae4?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [f1a5aa8a633636138fededdf46a8dcc20f899a48](https://github.com/cockroachdb/cockroach/commits/f1a5aa8a633636138fededdf46a8dcc20f899a48):


```
=== RUN   TestLogicalJobResiliency
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogicalJobResiliency3980847195
    test_log_scope.go:81: use -show-logs to present logs inline
    logical_replication_job_test.go:860: waiting for replication job 996672310541058049
    logical_replication_job_test.go:1033: condition failed to evaluate within 3m45s: from logical_replication_job_test.go:1037: stream ingestion has not recorded any progress yet, waiting to advance pos 1724230651.315033530,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogicalJobResiliency3980847195
--- FAIL: TestLogicalJobResiliency (346.04s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogicalJobResiliency.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/crosscluster/logical.TestLogicalJobResiliency [failed](https://tanzanite.cluster.engflow.com/invocations/default/57b682fd-3e16-4b8a-a008-945008aad225?testReportRun=15&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [18c4e23306bb99cc28c83bb3e5726f6f4279b798](https://github.com/cockroachdb/cockroach/commits/18c4e23306bb99cc28c83bb3e5726f6f4279b798):


```
=== RUN   TestLogicalJobResiliency
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogicalJobResiliency3086443816
    test_log_scope.go:81: use -show-logs to present logs inline
    logical_replication_job_test.go:924: waiting for replication job 997023699497025537
    logical_replication_job_test.go:1097: condition failed to evaluate within 45s: from logical_replication_job_test.go:1101: stream ingestion has not recorded any progress yet, waiting to advance pos 1724337886.434387137,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogicalJobResiliency3086443816
--- FAIL: TestLogicalJobResiliency (72.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=15</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogicalJobResiliency.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/crosscluster/logical/logical_replication_job_test.go
+++ b/pkg/ccl/crosscluster/logical/logical_replication_job_test.go
@@ -833,6 +833,7 @@ func TestLogicalJobResiliency(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderRace(t, "multi cluster/node config exhausts hardware")
+	skip.UnderDeadlock(t, "Scattering prior to creating LDR job slows down ingestion")
 
 	clusterArgs := base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{
@@ -1002,7 +1003,7 @@ func CreateScatteredTable(t *testing.T, db *sqlutils.SQLRunner, numNodes int, db
 	// ranges, so if we write just a few ranges those might all be on a single
 	// server, which will cause the test to flake.
 	numRanges := 50
-	rowsPerRange := 40
+	rowsPerRange := 20
 	db.Exec(t, "INSERT INTO tab (pk) SELECT * FROM generate_series(1, $1)",
 		numRanges*rowsPerRange)
 	db.Exec(t, "ALTER TABLE tab SPLIT AT (SELECT * FROM generate_series($1::INT, $2::INT, $3::INT))",

